includes/Core/Admin/Screen.php

```diff
@@ -129,9 +130,57 @@ public function add( Context $context ) {
 						$this->args['capability'],
 						$this->slug,
 						'',
-						$context->url( 'dist/assets/images/logo-g_white_small.png' )
+						'data:image/svg+xml;base64,' . Google_Icon::to_base64()
 					);
 					$menu_slug = $this->slug;
+
+					/**
+					 * An SVG icon file needs to be colored (filled) based on the theme color setting.
+					 *
+					 * This exists in js as wp.svgPainter() per:
+					 * https://github.com/WordPress/WordPress/blob/master/wp-admin/js/svg-painter.js
+					 *
+					 * The downside of the js approach is that we get a brief flash of an unstyled icon
+					 * until the JS runs.
+					 *
+					 * A user can pick a custom Admin Color Scheme, which is only available in admin_init
+					 * or later actions. add_menu_page runs on the admin_menu action, which precedes admin_init
+					 * per https://codex.wordpress.org/Plugin_API/Action_Reference
+					 *
+					 * WordPress provides some color schemes out of the box, but they can also be added via
+					 * wp_admin_css_color()
+					 *
+					 * Our workaround is to set the icon and subsequently replace it in admin_init, which is
+					 * what we do in the following action.
+					 */
+					add_action(
+						'admin_init',
+						function() {
+							global $menu, $_wp_admin_css_colors, $pagenow;
+							$color_scheme = get_user_option( 'admin_color' ) ?: 'fresh';
+
+							$prefix = 'googlesitekit-';
+
+							// If we're on one of the sitekit pages, use the 'current' color, otherwise use the 'base' color.
+							// @see wp_admin_css_color().
+							$color_key = 'admin.php' === $pagenow && substr( filter_input( INPUT_GET, 'page', FILTER_SANITIZE_STRING ), 0, strlen( $prefix ) ) === $prefix ? 'current' : 'base';
```

Comment:
@aaemnnosttv a new addition here as the following two URLs will have different fill colours:

`/wp-admin/admin.php?page=googlesitekit-dashboard` and `/wp-admin/index.php`

Reply:
Thanks, I noticed that in my testing too. Can we base this on `get_current_screen` instead of a query param? Assuming that the timing isn't an issue, we should simply be able to compare the Screen's `parent_file` or `parent_base` 
![image](https://user-images.githubusercontent.com/1621608/109281821-74955980-7825-11eb-9a4e-7f643c78a178.png)

Reply:
Tried that, however:

/wp-admin/admin.php?page=googlesitekit-settings
```
["base"]=>
string(36) "site-kit_page_googlesitekit-settings"
["parent_base"]=>
NULL
["parent_file"]=>
NULL
```

/wp-admin/admin.php?page=googlesitekit-dashboard
```
["base"]=>
string(37) "toplevel_page_googlesitekit-dashboard"
["parent_base"]=>
NULL
["parent_file"]=>
NULL
```

/wp-admin/admin.php?page=googlesitekit-module-analytics
```
["base"]=>
string(44) "site-kit_page_googlesitekit-module-analytics"
["parent_base"]=>
NULL
["parent_file"]=>
NULL
```

If there's an aversion to using `$_GET['page']`, we can use get_current_screen and see if the `base` or `id` params contain `googlesitekit`. It does mean we hook `current_screen` instead of `admin_init` but that doesn't break any functionality.

Reply:
@ivankruchkoff Ah yes, I see the current screen isn't set yet when `admin_init` fires.

> If there's an aversion to using `$_GET['page']`, we can use get_current_screen and see if the `base` or `id` params contain `googlesitekit`. It does mean we hook `current_screen` instead of `admin_init` but that doesn't break any functionality.

We should prefer to use the screen where possible over query parameters directly. From what I can see, `get_current_screen()->parent_base === 'googlesitekit-dashboard'` will be true on all of our screens with one exception (which is a bit of an edge case). When visiting the splash screen when already authenticated (we don't actually do this) the splash screen is registered as an "orphaned" admin page (not a child of our dashboard); see #2529. We talked about this yesterday but I think we may fix this by introducing a redirect to the dashboard in this case so it may be a non-issue.

So to answer your question, I think it's fine to hook onto `current_screen` to do the same thing since it's before the menu is generated/output, but we should be able to be more precise than simply checking a substring match. We should check that the parent base matches.

Reply:
If you take a look at my previous comment, `parent_base` is null in most scenarios.

Reply:
@ivankruchkoff It's not that I didn't look, but it seemed to be set in my testing but I think it appeared that way due to my test method and those being populated later on the instance. Logging it out differently now I see what you mean.

This works however, although I'm not sure what the ramifications of calling this earlier would be:
```php
add_action( 'current_screen', function ( WP_Screen $screen ) {
	global $parent_file;
	$screen->set_parentage( $parent_file );
	// $screen->parent_base is now set
} );
```

Normally, this isn't called until [after the menu is generated](https://github.com/WordPress/WordPress/blob/c504140c321776860fbb48703c36415e933c7e70/wp-admin/admin-header.php#L259) but I'm not sure why 🤔 

This is getting a bit too far down the rabbit hole though so if it's not possible without further hackiness then I'm happy to fallback to the query param-based logic if that's cleanest.

**One other screen-based alternative we could try** would be using the `$screen->id` which _is_ set on `current_screen` and should match the return value of `add_submenu_page` or `add_menu_page` in `Screen::add()` which is where we're already making changes. What do you think?

Reply:
Updated to use `get_current_screen()->id` per our sync and it also covers the dev/tester plugin pages too.